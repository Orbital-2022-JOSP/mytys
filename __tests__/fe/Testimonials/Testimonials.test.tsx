import { render } from '@testing-library/react';
import { Testimonials } from '../../../components/Testimonials/Testimonials';

jest.mock("next-auth/react", () => {
    const originalModule = jest.requireActual('next-auth/react');
    const mockSession = {};
    return {
        __esModule: true,
        ...originalModule,
        useSession: jest.fn(() => {
            return { data: mockSession, status: 'unauthenticated' }
        }),
    };
});

describe("The Testimonials Component", () => {

    it('Should render correctly',
        async () => {
            const { container } = render(<Testimonials />);

            expect(container).toMatchSnapshot()
        })
})

