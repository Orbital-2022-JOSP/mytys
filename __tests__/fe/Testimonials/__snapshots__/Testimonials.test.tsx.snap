// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Testimonials Component Should render correctly 1`] = `
<div>
  <div
    class="px-4 py-16 mx-auto sm:max-w-xl md:max-w-full lg:max-w-screen-xl sm:min-w-xl md:min-w-full lg:min-w-screen-lg xl:min-w-screen-xl md:px-24 lg:px-8 lg:py-20"
  >
    <h2
      class="max-w-xl mb-10 font-sans text-3xl font-bold leading-none text-center text-light-blue-500 tracking-tight text-gray-900 sm:text-4xl md:mx-auto"
    >
      What Our Students Say
    </h2>
    <div
      class="grid gap-8 lg:grid-cols-3 sm:max-w-sm sm:mx-auto lg:max-w-full"
    >
      <div
        class="overflow-hidden transition-shadow duration-300 bg-white rounded shadow-sm"
      >
        <img
          alt=""
          class="object-cover w-full h-64"
          src="images/increase_trend_graph_img.webp"
        />
        <div
          class="p-5 border border-t-0"
        >
          <p
            class="mb-3 text-xs font-semibold tracking-wide uppercase"
          >
            <span
              class="text-blue-gray-900"
            >
              Samuel  
              <span
                class="text-gray-600"
              >
                — Primary 4 Science
              </span>
            </span>
          </p>
          <a
            aria-label="Category"
            class="inline-block mb-3 text-2xl font-bold leading-5 transition-colors duration-200 hover:text-light-blue-500"
            href="/"
            title="Visit the East"
          >
            From C to A!
          </a>
          <p
            class="mb-2 text-gray-700"
          >
            I never would have imagined to able to jump from a C grade to an A grade! Within 3 months of using MYTYS, I could finally achieve my academic dreams!
          </p>
          <a
            aria-label=""
            class="inline-flex items-center font-semibold transition-colors duration-200 text-light-blue-500 hover:text-light-blue-800"
            href="/"
          >
            Read more
          </a>
        </div>
      </div>
      <div
        class="overflow-hidden transition-shadow duration-300 bg-white rounded shadow-sm"
      >
        <img
          alt=""
          class="object-cover w-full h-64"
          src="images/A_grade_img.webp"
        />
        <div
          class="p-5 border border-t-0"
        >
          <p
            class="mb-3 text-xs font-semibold tracking-wide uppercase"
          >
            <span
              class="text-blue-gray-900"
            >
              John  
              <span
                class="text-gray-600"
              >
                — Primary 3 Math
              </span>
            </span>
          </p>
          <a
            aria-label="Category"
            class="inline-block mb-3 text-2xl font-bold leading-5 transition-colors duration-200 hover:text-light-blue-500"
            href="/"
            title="Visit the East"
          >
            4A's in my recent exam!
          </a>
          <p
            class="mb-2 text-gray-700"
          >
            With just 1 month of using MYTYS, I was able to excel in my subjects this school term. I believe it will help me in my grades in future terms as well!
          </p>
          <a
            aria-label=""
            class="inline-flex items-center font-semibold transition-colors duration-200 text-light-blue-500 hover:text-light-blue-800"
            href="/"
          >
            Read more
          </a>
        </div>
      </div>
      <div
        class="overflow-hidden transition-shadow duration-300 bg-white rounded shadow-sm"
      >
        <img
          alt=""
          class="object-cover w-full h-64"
          src="images/lightbulb_img.jpg"
        />
        <div
          class="p-5 border border-t-0"
        >
          <p
            class="mb-3 text-xs font-semibold tracking-wide uppercase"
          >
            <span
              class="text-blue-gray-900"
            >
              Joshua
              <span
                class="text-gray-600"
              >
                — Primary 6 Math
              </span>
            </span>
          </p>
          <a
            aria-label="Category"
            class="inline-block mb-3 text-2xl font-bold leading-5 transition-colors duration-200 hover:text-light-blue-500"
            href="/"
            title="Visit the East"
          >
            Learning is Fun with MYTYS
          </a>
          <p
            class="mb-2 text-gray-700"
          >
            I used to hate mathematics since Primary 1, but the gamified platform on MYTYS enabled me to learn math and have fun at the same time!
          </p>
          <a
            aria-label=""
            class="inline-flex items-center font-semibold transition-colors duration-200 text-light-blue-500 hover:text-light-blue-800"
            href="/"
          >
            Read more
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
